### ISSUE TITLE
com.hazelcast.partition.PartitionMigrationListenerTest.testMigrationStats_afterPromotions [HZ-3556]

### ISSUE THREAD

**Issue by Patras3** [hazelcast/hazelcast#24708]
_master_ (commit 120d25873b173dbc3ec157b1d68c3980168978ef)

Failed on redhat-17: https://jenkins.hazelcast.com/job/Hazelcast-master-RedHatJDK17/67/testReport/junit/com.hazelcast.partition/PartitionMigrationListenerTest/testMigrationStats_afterPromotions/
<details><summary>Stacktrace:</summary>

```
java.lang.AssertionError: expected:<181> but was:<6>
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotEquals(Assert.java:835)
	at org.junit.Assert.assertEquals(Assert.java:647)
	at org.junit.Assert.assertEquals(Assert.java:633)
	at com.hazelcast.partition.PartitionMigrationListenerTest.assertMigrationProcessEventsConsistent(PartitionMigrationListenerTest.java:197)
	at com.hazelcast.partition.PartitionMigrationListenerTest.testMigrationStats_afterPromotions(PartitionMigrationListenerTest.java:180)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:115)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:107)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:833)

```
</details>

Standard output can be found here - https://s3.console.aws.amazon.com/s3/buckets/j-artifacts/Hazelcast-master-RedHatJDK17/67/

**Comment by Patras3** [comment]
Failed again on _master_ with Windows - MicrosoftJDK11: https://jenkins.hazelcast.com/job/Hazelcast-master-Windows-MicrosoftJDK11/69/testReport/com.hazelcast.partition/PartitionMigrationListenerTest/testMigrationStats_afterPromotions/

**Comment by JamesHazelcast** [comment]
This issue is potentially related to, and resolved by, this PR: https://github.com/hazelcast/hazelcast/pull/25114

**Comment by olukas** [comment]
Failed again on _master_ with openjdk-17: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-ZuluJDK17/108/testReport/com.hazelcast.partition/PartitionMigrationListenerTest/testMigrationStats_afterPromotions/

**Comment by olukas** [comment]
Reopening as issue occurred again on: _5.3.z_ (commit d160431eb4c8b7e22ede5f266274caed2b82bab3)
with RedHatJDK11: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-5.maintenance-RedHatJDK11/123/testReport/com.hazelcast.partition/PartitionMigrationListenerTest/testMigrationStats_afterPromotions/

@JamesHazelcast could you please backport the fix to `5.3.z` branch?

<details><summary>Stacktrace:</summary>

```
java.lang.AssertionError: expected:<179> but was:<5>
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotEquals(Assert.java:835)
	at org.junit.Assert.assertEquals(Assert.java:647)
	at org.junit.Assert.assertEquals(Assert.java:633)
	at com.hazelcast.partition.PartitionMigrationListenerTest.assertMigrationProcessEventsConsistent(PartitionMigrationListenerTest.java:202)
	at com.hazelcast.partition.PartitionMigrationListenerTest.testMigrationStats_afterPromotions(PartitionMigrationListenerTest.java:185)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:115)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:107)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:829)

```
</details>



Standard output can be found here - https://s3.console.aws.amazon.com/s3/buckets/j-artifacts/Hazelcast-5.maintenance-RedHatJDK11/123/

**Comment by AyberkSorgun** [comment]
Fixed in https://github.com/hazelcast/hazelcast-mono/pull/754
